### PR TITLE
Add ability to use local embeddings model

### PR DESCRIPTION
**Pull request by Tymec**
### Background

This change adds the ability to use a local embeddings model (sBERT) using the SentenceTransformers library. This change gives the users an alternative to OpenAI's ada embeder which costs money.

### Changes

- Renamed the `get_ada_embedding` function to `get_embedding`
- `get_embedding` function now uses either **ada** or **sBERT** embeder based on `memory_embeder` config option
- New config option called `memory_embeder` that defaults to **ada**, which can also be configured by setting the MEMORY_EMBEDER environment variable.
- Added `sentence_transformers` as a requirement

### Documentation

- In-code comments that explain the logic and functionality of the code.

### Test Plan

- Created a unit test module to test if everything works as intended.

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by FlyWieEinAirplane** [comment]
Sounds useful.
Looking at the code, it would be easy to add a third option but it seems to be a bit more complicated to add a third embeddings dimension option. Maybe this could be improved a bit.

File: .env.template [hunk 1]
```diff
@@ -50,6 +50,8 @@ SMART_TOKEN_LIMIT=8000
 
 # MEMORY_BACKEND - Memory backend type (Default: local)
 MEMORY_BACKEND=local
+# MEMORY_EMBEDER - Embeddings model to use (Default: ada)
+MEMORY_EMBEDER=ada
```
Comment (nponeccop) [inline .env.template]: EMBEDDER is the right spelling I guess, double D as in EMBEDDED

File: tests/embeder_test.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    def test_sbert(self):
+        self.cfg.memory_embeder = "sbert"
+        text = "Sample text"
+        result = get_embedding(text)
+        self.assertEqual(result.shape, (768,))
+
+
+if __name__ == '__main__':
+    unittest.main()
```
Comment (nponeccop) [inline tests/embeder_test.py]: We need the final line feed (as per pep8). Just press Enter once afrter main() and save

**Review by nponeccop — CHANGES_REQUESTED** [review]

**Comment by Tymec** [comment]
Thanks for the feedback. I agree that it's not a good approach and only did so as a temporary solution which I overlooked.

To improve it, I created a variable that holds a dictionary with the embeder name as the key and the dimension as the value. This variable is dynamically assigned to the dimension based on the embeder used at runtime. Then, this variable is imported and used in memory backends.

File: .env.template (hunk 1 above)
Comment (Tymec) [reply .env.template]: Should be fixed with 2a147ac and 34eac57.

File: tests/embeder_test.py (hunk 2 above)
Comment (Tymec) [reply tests/embeder_test.py]: Fixed with 34eac57.

**Review by nponeccop — APPROVED** [review]
There is a crazy EOF error now too. I'm approving.

**Comment by Tymec** [comment]
Just noticed and patched some issues with the test unit, as well as resolved issues with the new EMBED_DIM variable in memory.base module.

**Comment by nponeccop** [comment]
@Tymec There are conflicts now

**Comment by nponeccop** [comment]
@Tymec There are conflicts now

**Review by nponeccop — APPROVED** [review]

**Review by richbeales — APPROVED** [review]

**Comment by richbeales** [comment]
Run flake8 autogpt/ tests/ --select E303,W293,W291,W292,E305,E231,E302
autogpt/config/config.py:85:1: W293 blank line contains whitespace
autogpt/memory/base.py:46:1: W293 blank line contains whitespace
autogpt/memory/base.py:48:1: W293 blank line contains whitespace

**Review by nponeccop — APPROVED** [review]

**Pull request merged**